students/tri_nguyen/lesson03/slicing.py

```diff
@@ -0,0 +1,144 @@
+# -------------------------------------------#
+# Student:   Tri Nguyen
+# Instructor: Natasha Aleksandrova
+# Date:  08-Feb-2018
+# -------------------------------------------#
+
+# Write some functions that take a sequence as an
+# argument, and return a copy of that sequence:
+# 1. With the first and the last item exchanged
+# 2. With every other item removed
+# 3. With the first 4 and the last 4 items removed
+# and then every other item in between
+# 4. With the elements reversed (just slicing)
+# 5. With the middle third, the last third, then
+# the first third in the new order
+
+# Part 1:
+
+def exchange_first_last(seq):
+    '''
+        this function takes a sequence (seq) and return
+        a copy with the first and the last item exchanged        
+    '''
+    if len(seq) <= 1:
+        exchanged_seq = 'Sequence has 0 or 1 value. Unable to compute.'
+    else:
+        exchanged_seq = seq[-1:] + seq[1:len(seq) - 1] + seq[:1]
+    return exchanged_seq
```

Comment:
your slicing logic is great, however it's totally valid for a function like this to return an empty list if empty list is given.

so your function could be super simple: 
```
def exchange_first_last(seq):
   return seq[-1:] + seq[1:len(seq) - 1] + seq[:1]
```

same comment applies to other functions